### PR TITLE
OpenAI refactor [2/n]: Replace structured rag events with unstructured

### PR DESCRIPTION
**Pull request by rossdanlm**
OpenAI refactor [2/n]: Replace structured rag events with unstructured

This breaks when using it multiple times. Using the unstructured event now

### Before
No data in RagEvent table
Screenshot 2024-05-29 at 16 11 53

Data in the RagQueryTrace table
Screenshot 2024-05-29 at 16 10 52

Jaeger collector trace: http://44.222.237.105:16686/trace/1dc2a26a96c32453a494922d4d20d0fb

### After

Data in RagEvent table
Screenshot 2024-05-29 at 16 12 28

No data in the RagQueryTrace table
Screenshot 2024-05-29 at 16 08 34

Jaeger collector trace: http://44.222.237.105:16686/trace/29c5a46ce4a434b1c469fc20a54b401b

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/tracing_auto_instrumentation/pull/32).
* #33
* #36
* #35
* __->__ #32

### CODE REVIEW

File: src/tracing_auto_instrumentation/openai/openai.py [hunk 1]
```diff
@@ -102,52 +93,61 @@ def gen():
...
+                        "completion_tokens"
+                    ],
+                    "choices": json_serialize_anything(
+                        log_response["choices"]
+                    ),
+                    **params_flat,
+                }
+            )
+            try:
+                output = log_response["choices"][0]["message"]["content"]
```
Comment (rossdanlm) [inline src/tracing_auto_instrumentation/openai/openai.py:131]: all re-indent formatting

File: src/tracing_auto_instrumentation/openai/openai.py [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
-from typing import Any, Mapping
+from typing import Any, Mapping, Optional
 import openai as openai_module
 
-from lastmile_eval.rag.debugger.tracing.lastmile_tracer import LastMileTracer
-from lastmile_eval.rag.debugger.common.query_trace_types import (
-    LLMOutputReceived,
-    PromptResolved,
-)
+from lastmile_eval.rag.debugger.api import LastMileTracer
```
Comment (rossdanlm) [inline src/tracing_auto_instrumentation/openai/openai.py:6]: This is a crucial change. Python is a bad language so you can import however you want, but we should be importing from the actual places where we say we should import the modules. Doing so from the api class provides all the api methods which means you can auto-complete stuff in your IDE

**Comment by rholinshead** [comment]
Shouldn't RAGQueryTrace table have input/output and eventData ?

**Comment by rossdanlm** [comment]
> Shouldn't RAGQueryTrace table have input/output and eventData ?

Yes they do have input and output and eventData, but I just prefer to write to the RagEvent table. We automatically "write" to the RagQueryTrace table at the end of the span. I put write in quotes because we do this even if the data to write to it was empty and we never called any input or output. But even more so, having input, output, eventData for a trace level doesn't make sense since you can have multi-turn and multiple spans in the openAI wrapper, so you don't really know what the "trace" level data really is, which is why all the other auto-instrumentors use the unstructured events too

**Pull request merged**